pkg/serverless/daemon/daemon_test.go

```diff
@@ -124,24 +126,10 @@ func TestTellDaemonRuntimeDoneOnceStartAndEndAndTimeout(t *testing.T) {
 func TestRaceTellDaemonRuntimeStartedVersusTellDaemonRuntimeDone(t *testing.T) {
 	port := testutil.FreeTCPPort(t)
 	d := StartDaemon(fmt.Sprint("127.0.0.1:", port))
-	time.Sleep(100 * time.Millisecond)
 	defer d.Stop()
 
-	d.TellDaemonRuntimeStarted()
-
-	go func() {
-		for i := 0; i < 1000; i++ {
-			go d.TellDaemonRuntimeStarted()
-		}
-	}()
-
-	go func() {
-		for i := 0; i < 1000; i++ {
-			go d.TellDaemonRuntimeDone()
-		}
-	}()
-
-	time.Sleep(2 * time.Second)
+	go d.TellDaemonRuntimeStarted()
+	go d.TellDaemonRuntimeDone()
```

Comment:
This test rewrite will still fail if the race condition we were testing for ever returns.